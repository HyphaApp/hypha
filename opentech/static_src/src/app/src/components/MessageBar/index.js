import React from 'react'
import PropTypes from 'prop-types'

import { MESSAGE_TYPES } from '@actions/messages'

const MessageBar = ({message, onDismiss}) => {

    return (
        <div className={message.type}>
            <p>{message.message}</p>
            {onDismiss && <button onClick={onDismiss}>[X]</button>}
        </div>
    )
}

MessageBar.propTypes = {
    message: PropTypes.shape({
        type: PropTypes.oneOf(Object.values(MESSAGE_TYPES)),
        message: PropTypes.string
    }),
    onDismiss: PropTypes.func,
}

export default MessageBar
