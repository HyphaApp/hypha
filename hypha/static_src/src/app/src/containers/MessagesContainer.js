import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import MessageBar from '@components/MessageBar'
import MessagesList from '@components/MessagesList'
import { getMessages } from '@selectors/messages'
import { dismissMessage } from '@actions/messages'

export const MessagesContainer = ({ messages, dismiss }) => {
    return (
        <MessagesList>
            {Object.values(messages).map(({ message, type, id}) =>
                <MessageBar key={id} message={message} type={type}
                    onDismiss={() => dismiss(id)} />
            )}
        </MessagesList>
    )
}

const mapStateToProps = state => ({
    messages: getMessages(state),
})

const mapDispatchToProps = dispatch => ({
    dismiss: id => dispatch(dismissMessage(id)),
})

MessagesContainer.propTypes = {
    messages: PropTypes.object,
    dismiss: PropTypes.func,
}

export default connect(mapStateToProps, mapDispatchToProps)(MessagesContainer)
