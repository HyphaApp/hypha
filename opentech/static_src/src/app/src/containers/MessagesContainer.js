import React from 'react'
import { connect } from 'react-redux'

import MessageBar from '@components/MessageBar'
import { getMessages } from '@selectors/messages'
import { dismissMessage } from '@actions/messages'

const MessagesContainer = ({ messages, dismiss }) => {
    return Object.values(messages).map(v =>
        <MessageBar key={v.id} message={v}
            onDismiss={() => dismiss(v.id)} />
    )
}

const mapStateToProps = state => ({
    messages: getMessages(state),
})

const mapDispatchToProps = dispatch => ({
    dismiss: id => dispatch(dismissMessage(id)),
})

export default connect(mapStateToProps, mapDispatchToProps)(MessagesContainer)
