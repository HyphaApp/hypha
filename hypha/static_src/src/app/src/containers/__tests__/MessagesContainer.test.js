import React from "react";
import {mount} from "enzyme";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({adapter: new Adapter()});
import { MessagesContainer } from '../MessagesContainer'
import MessageBar from '@components/MessageBar'

describe("Test messages Container", () => {
    it("Should render messages outcome", () => {
        const messages = {m : {message : "message1", type : "success", id : 1}}
        const dismiss = jest.fn()
        const wrapper = mount(
                <MessagesContainer 
                messages={messages}
                dismiss={dismiss}
                />
        );
        wrapper.find('button').props().onClick()
        expect(dismiss).toHaveBeenCalled()
        expect(dismiss).toHaveBeenCalledWith(1)
        expect(wrapper.containsMatchingElement(<MessageBar />)).toBe(true)
        expect(wrapper).toMatchSnapshot();
    });
});