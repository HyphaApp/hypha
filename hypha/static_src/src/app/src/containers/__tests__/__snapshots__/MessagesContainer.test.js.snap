// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test messages Container Should render messages outcome 1`] = `
<MessagesContainer
  dismiss={
    [MockFunction] {
      "calls": Array [
        Array [
          1,
        ],
      ],
    }
  }
  messages={
    Object {
      "m": Object {
        "id": 1,
        "message": "message1",
        "type": "success",
      },
    }
  }
>
  <MessagesList>
    <ul
      className="messages"
    >
      <MessageBar
        key="1"
        message="message1"
        onDismiss={[Function]}
        type="success"
      >
        <li
          className="messages__text messages__text--success"
        >
          <div
            className="messages__inner"
          >
            <p
              className="messages__copy"
            >
              message1
            </p>
            <button
              className="button messages__button"
              onClick={[Function]}
            >
              OK
            </button>
          </div>
        </li>
      </MessageBar>
    </ul>
  </MessagesList>
</MessagesContainer>
`;
